### ISSUE TITLE
Odd state report

### ISSUE THREAD

**Issue by Transisto** [sparrowwallet/sparrow#464]
The wallet seems stuck like usual since 3 days ago (the bottom connection toggle is slowly blinking)

But I've seen a 2-3 mix get done yesterday by looking from a watch only electrum.

Then these connection log.

```
[Mar 09 22:07:13] Loading transactions for ../0/785
[Mar 09 22:07:13] Loading transactions for ../0/787
[Mar 09 22:07:13] Loading transactions for ../0/790
[Mar 09 22:07:13] Loading transactions for ../0/791
[Mar 09 22:07:13] Loading transactions for ../0/792
[Mar 10 19:32:50] Failed to retrieve transaction history for path: ../0/792
[Mar 11 06:11:46] Loading transactions for ../0/1310
[Mar 11 06:13:55] Loading transactions for ../0/1311
[Mar 11 14:29:10] Loading transactions for ../0/1313
[Mar 11 15:55:35] Loading transactions for ../0/1314
[Mar 12 00:37:10] Loading transactions for ../0/1315
[Mar 12 09:29:11] Loading transactions for ../0/1316
[Mar 12 10:56:34] Loading transactions for ../0/1317
```

It's odd to have for example a 9 hours gap between 
[Mar 12 00:37:10] Loading transactions for ../0/1315
[Mar 12 09:29:11] Loading transactions for ../0/1316

**Comment by Transisto** [comment]
What wallet doesn't close, process stay in background, Process dump is 1.2 GB Is this what you'd like uploaded for review?

**Comment by craigraw** [comment]
> Process dump file is 1.2 GB Is this what you'd like uploaded for review?

No, not the process dump, but rather the thread dump from `jstack` as described in https://github.com/sparrowwallet/sparrow/issues/457#issuecomment-1061624318. It's a few hundred lines of text.

What server are you connected to in this case? It appears to be blockstream's (the only one that doesn't support batching). Does this happen on any other public server?

**Comment by Transisto** [comment]
No matter what server I initially choose it end up cycling through all the public ones whenever there's a connection issue.

Connection ends up dropping near the end of a long sync.

I should have posted more or the transaction load logs.

Each deposit/postmix is configured to check as few trailing addresses as possible.

**Comment by Transisto** [comment]
[Mar 14 15:54:28] Finding transactions for ../1/2591
[Mar 14 15:54:28] Finding transactions for ../1/2592
[Mar 14 15:54:29] Finished loading.
[Mar 14 18:27:15] Finding transactions for [../0/0-../0/2616]
[Mar 14 18:30:55] Failed to subscribe to paths: [../0/0-../0/2616]
[Mar 15 01:27:12] Loading transactions for ../0/1319
[Mar 15 11:52:05] Loading transactions for ../0/1320
[Mar 15 13:01:41] Loading transactions for ../0/1321
[Mar 15 14:00:43] Loading transactions for ../0/1322
[Mar 15 19:46:24] Loading transactions for ../0/1323
[Mar 15 23:39:17] Loading transactions for ../0/1324
[Mar 16 10:44:48] Loading transactions for ../0/1325
[Mar 16 11:11:42] Loading transactions for ../0/1326
[Mar 16 12:12:18] Loading transactions for ../0/1331
[Mar 17 00:57:52] Loading transactions for ../0/1335
[Mar 17 01:00:59] Loading transactions for ../0/1336
[Mar 17 01:23:29] Loading transactions for ../0/1337
[Mar 17 02:00:24] Loading transactions for ../0/1338
[Mar 17 05:27:06] Loading transactions for ../0/1339
[Mar 17 06:54:06] Loading transactions for ../0/1340
[Mar 17 12:38:10] Loading transactions for ../0/1341
[Mar 17 13:00:32] Loading transactions for ../0/1342
[Mar 17 19:08:57] Loading transactions for ../0/1347
[Mar 18 04:42:32] Loading transactions for ../0/1348

Pointing  out to the timestamp, when it become slow.
Hopefully the 1.6.2 helps with this.

**Comment by craigraw** [comment]
This is indeed odd, but I'm not sure is there is much Sparrow can do practically to improve things. Clearly, all ElectrumX-based public servers have banned your IP due to their resource limiting configuration. The remaining Blockstream server, based on electrs-esplora, does not have the same resource limiting configuration (but also does not yet support JSON-RPC batching which will make loading deep wallets slow).

Its behaviour is odd in that it is keeping the socket open for a long period without delivering much in the way of data. This means Sparrow does not enter normal socket timeout behaviour, but also doesn't get to move forward with wallet loading for several hours. I think this behaviour is better addressed on the electrs-esplora project - probably the server should be more decisive on whether it wants to keep serving data, but I'm unclear on why it slows down so much in the first place.
